### PR TITLE
SWP-1977 - change extra from serialized string to relations 

### PR DESCRIPTION
**Pull request by greg606**
Fixes #
SWP-1977

## Proposed Changes

Adds Article relation Extrafields replacing serialized extra field
Based on change to metadata field here: https://github.com/superdesk/web-publisher/pull/1123

License: AGPLv3

### CODE REVIEW

File: src/SWP/Bundle/ContentBundle/Model/Article.php [hunk 1]
```diff
@@ -446,4 +481,71 @@ public function setMetadata(array $metadata): void
...
+        }
+
+        foreach ($this->getExtraEmbedFields() as $extra) {
+            $this->removeExtraEmbedFields($extra);
+        }
+
+        foreach ($extra as $key => $value) {
+
+            if(is_array($value)) {
+                $embed = new ArticleExtraEmbedField();
```
Comment (takeit) [inline src/SWP/Bundle/ContentBundle/Model/Article.php:538]: You can create `named constructors` in `ArticleExtraEmbedField` and `ArticleExtraTextField` classes, so this part of the code is more readable, and we will have a one way to construct those objects.

**Review by takeit — CHANGES_REQUESTED** [review]
Please, also run the PHP CS fixer on files you made changes to

**Review by takeit — APPROVED** [review]

File: src/SWP/Bundle/ContentBundle/Resources/config/serializer/Model.Article.yml [hunk 2]
```diff
@@ -1,5 +1,7 @@
 SWP\Bundle\ContentBundle\Model\Article:
     exclusion_policy: ALL
+    accessor_order: custom
+    custom_accessor_order: [id,title,body,slug,publishedAt,status,route,isPublishable,metadata,lead,code,sources,extra,slideshows,previousRelativeUrls,createdAt,updatedAt,authors,keywords,media,templateName,publishStartDate,publishEndDate,featureMedia,seoMetadata]
```
Comment (ahilles107) [inline src/SWP/Bundle/ContentBundle/Resources/config/serializer/Model.Article.yml:4]: why You want to maintain that order?

File: src/SWP/Bundle/CoreBundle/Migrations/2020/09/Version20200923162328.php [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        $this->abortIf(
+            'postgresql' !== $this->connection->getDatabasePlatform()->getName(),
+            'Migration can only be executed safely on \'postgresql\'.'
+        );
+
+        $this->addSql('DROP SEQUENCE swp_article_extra_id_seq CASCADE');
+        $this->addSql('DROP TABLE swp_article_extra');
+    }
+
+    public function postUp(Schema $schema): void
```
Comment (ahilles107) [inline src/SWP/Bundle/CoreBundle/Migrations/2020/09/Version20200923162328.php:60]: imo there it should be special command.

File: src/SWP/Bundle/ContentBundle/Resources/config/serializer/Model.Article.yml (hunk 2 above)
Comment (greg606) [reply src/SWP/Bundle/ContentBundle/Resources/config/serializer/Model.Article.yml:4]: because of behat tests requiring exact json content.

Comment (greg606) [reply src/SWP/Bundle/ContentBundle/Resources/config/serializer/Model.Article.yml:4]: probably would make more sense to improve these tests somehow?

Comment (ahilles107) [reply src/SWP/Bundle/ContentBundle/Resources/config/serializer/Model.Article.yml:4]: Yep, instead checking whole json we should check specific properties and their values.

**Comment by takeit** [comment]
@greg606 please take a look at the comments, let's merge it to `master` branch and not `2.1` branch.

**Pull request merged**